crates/iota-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -258,6 +405,17 @@ impl<'a> LimitsTraversal<'a> {
         Ok(())
     }
 
+    /// Error returned if transaction payloads exceed limit. Also sets the
+    /// transaction payload budget to zero to indicate it has been spent
+    /// (This is done to prevent future checks for smaller arguments from
+    /// succeeding even though a previous larger argument has already
+    /// failed).
+    fn tx_payload_size_error(&mut self) -> ServerError {
+        self.tx_payload_budget = 0;
+        self.reporter
+            .payload_size_error("Transaction payload too large")
```

Comment:
Could we also say what was the size of transaction payload? So the user knows how much he exceeded the value.

Reply:
No we can't. This is lazily evaluated, so we just know it exceeded the limit set by the service. Can be more explitit about it here though.